packages/next/src/client/components/router-reducer/reducers/restore-reducer.ts

```diff
@@ -4,13 +4,15 @@ import {
   ReducerState,
   RestoreAction,
 } from '../router-reducer-types'
+import { prunePrefetchCache } from './prune-prefetch-cache'
 
 export function restoreReducer(
   state: ReadonlyReducerState,
   action: RestoreAction
 ): ReducerState {
   const { url, tree } = action
   const href = createHrefFromUrl(url)
+  prunePrefetchCache(state.prefetchCache)
```

Comment:
I'm not sure if we have to do this in back, might be better to optimize for when `prefetch()` is triggered on the page instead, as back needs to be as fast as possible.

Reply:
Actually it makes more sense to do it there instead, you're right. I actually had issues because the effect might run before or after the prefetch call so it might not get pruned in time. Moving it to the prefetch reducer will solve it